csrc/multidevice/utils.cpp

```diff
@@ -53,9 +29,33 @@ std::vector<IterDomain*> getShardedIterDomains(TensorView* tv) {
       [](auto id) { return id->isDeviceDim(); });
   return sharded_ids;
 }
-
 } // namespace
 
+bool isSharded(TensorView* tv) {
+  bool is_sharded = false;
+  auto rids = TensorDomain::noReductions(tv->getMaybeRFactorDomain());
+  auto ids = TensorDomain::noReductions(tv->getLeafDomain());
+  for (auto i : c10::irange(ids.size())) {
+    // Only one axis can be sharded on DIDx.
+    NVF_ERROR(
+        !(is_sharded && ids[i]->isDeviceDim()),
+        "Multiple IterDomains parallelized on DIDx in TensorView ",
+        tv->toString());
+
+    if (ids[i]->isDeviceDim()) {
+      // Currently do not support split/merge on a device dimension.
+      auto exprs = DependencyCheck::getAllExprsBetween(
```

Comment:
Why not just 

```suggestion
      NVF_ERROR(std::find(rids.begin(), rids.end(), ids[i]) != rids.end());
```

Reply:
much cleaner! thanks!

Reply:
I'm probably confused, but shouldn't it be a `==` instead of `!=`?
Sorry if I'm wrong, anyway if it passes the tests it should be ok!